File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums):
        ans = 0
        cnt = 0
        l = 0
        
        for r in range(len(nums)):
            # Increase count if the current element is 0
            cnt += nums[r] ^ 1
            
            # If more than one zero, move the left pointer
            while cnt > 1:
                cnt -= nums[l] ^ 1
                l += 1
            
            # Calculate the length of the current valid subarray
            ans = max(ans, r - l)
        
        return ans

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.js
var longestSubarray = function(nums) {
    let prevWindow = 0;
    let currWindow = 0;
    let max_length = 0;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) {
            max_length = Math.max(max_length, prevWindow + currWindow);
            prevWindow = currWindow;
            currWindow = 0;
        } else {
            currWindow++;
        }
    }

    if (currWindow === nums.length) {
        return currWindow - 1;
    }

    max_length = Math.max(max_length, prevWindow + currWindow);
    return max_length;
};